aws-s3-transfer-manager/src/operation/download_objects/worker.rs

```diff
@@ -142,8 +142,12 @@ async fn download_single_obj(
     let delim = ctx.state.input.delimiter();
 
     let key_path = local_key_path(root_dir, key.as_str(), prefix, delim)?;
-    let mut handle =
-        crate::operation::download::Download::orchestrate(ctx.handle.clone(), input).await?;
+    let mut handle = crate::operation::download::Download::orchestrate(
+        ctx.handle.clone(),
+        input,
+        tracing::Span::current().id(),
```

Comment:
I don't think you need this, the current span is always used as the parent by default. I think you just need to call `.instrument()` on this future before calling `.await`. Or in this case if you want the parent to be the parent you can use [`in_current_span()`](https://docs.rs/tracing-futures/0.2.5/tracing_futures/trait.Instrument.html#method.in_current_span) for that.

```rs
    let mut handle = crate::operation::download::Download::orchestrate(
        ctx.handle.clone(),
        input)
     .in_current_span()
     .await?;
```

Reply:
You're right that I don't need to be passing the current span's ID. I changed this argument to a bool: `use_current_span_as_parent_for_tasks`

But we talked. I do still want something to control whether the tasks for single-object-download go under the current span, or becomes a new root span. I added some detailed comments about this...